src/AzureFunctions/Example/Startup.cs

```diff
@@ -0,0 +1,31 @@
+using GraphQL;
+using GraphQL.Server;
+using GraphQL.Types;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using StarWars;
+using StarWars.Types;
+
+[assembly: FunctionsStartup(typeof(Example.Startup))]
+
+namespace Example
+{
+    class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.Services.AddSingleton<StarWarsData>();
+            builder.Services.AddSingleton<StarWarsQuery>();
+            builder.Services.AddSingleton<StarWarsMutation>();
+            builder.Services.AddSingleton<HumanType>();
+            builder.Services.AddSingleton<HumanInputType>();
+            builder.Services.AddSingleton<DroidType>();
+            builder.Services.AddSingleton<CharacterInterface>();
+            builder.Services.AddSingleton<EpisodeEnum>();
+            builder.Services.AddSingleton<ISchema, StarWarsSchema>();
+
+            builder.Services.AddSingleton<IDocumentExecuter>(sp => new DocumentExecuter());
```

Comment:
```suggestion
            builder.Services.AddSingleton<IDocumentExecuter, DocumentExecuter>();
```

Reply:
I tried that and then I get an "Object reference not set to an instance of an object". I had not take a deep look into this.

Reply:
Interesting to look at stacktrace.

Reply:
I think the DI will prefer the constructors with more parameters (https://github.com/graphql-dotnet/graphql-dotnet/blob/493fbfc804ea9c5ce33d6dab939f02114ab91cc6/src/GraphQL/Execution/DocumentExecuter.cs#L29) and call it with null. I see to possible ways:
 - keep the lambda
 - register all 3 services

To not run into problems, if more dependency needed, I would prefer the lambda. What is your opinion?

Reply:
> I think the DI will prefer the constructors with more parameters 

It is true if **all** ctor deps are registered in container. Does azure app have different di container? If you look into graphql-dotnet Harness project you see the same di configuration which works and normally calls default `DocumentExecutor` ctor. Interesting to look at error stacktrace.

Reply:
The Stack Trace does contain not much:

```
   at GraphQL.DocumentExecuter.<ExecuteAsync>d__7.MoveNext() in C:\Users\sschoof\Documents\GitHub\graphql-dotnet\src\GraphQL\Execution\DocumentExecuter.cs:line 92
```

(Is this line https://github.com/graphql-dotnet/graphql-dotnet/blob/dfa171446e7a6d964c7d8281d52e2b58783c73c0/src/GraphQL/Execution/DocumentExecuter.cs#L92)

I added a break point to the constructor and see that the arguments are null. The Stack Trace at the constructor:
```
   at System.Environment.get_StackTrace()
   at GraphQL.DocumentExecuter..ctor(IDocumentBuilder documentBuilder, IDocumentValidator documentValidator, IComplexityAnalyzer complexityAnalyzer) in C:\Users\sschoof\Documents\GitHub\graphql-dotnet\src\GraphQL\Execution\DocumentExecuter.cs:line 34
   at lambda_method(Closure , IResolverContext )
   at DryIoc.Factory.<>c__DisplayClass26_1.<ApplyReuse>b__2() in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 6596
   at DryIoc.Scope.TryGetOrAdd(ImMap`1 items, Int32 id, CreateScopedValue createValue, Int32 disposalOrder) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 7840
   at DryIoc.Scope.GetOrAdd(Int32 id, CreateScopedValue createValue, Int32 disposalOrder) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 7825
   at DryIoc.Factory.ApplyReuse(Expression serviceExpr, Request request) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 6595
   at DryIoc.Factory.GetExpressionOrDefault(Request request) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 6555
   at DryIoc.ReflectionFactory.CreateExpressionOrDefault(Request request) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 7074
   at DryIoc.Factory.GetExpressionOrDefault(Request request) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 6545
   at DryIoc.Factory.GetDelegateOrDefault(Request request) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 6625
   at DryIoc.Container.ResolveAndCacheDefaultFactoryDelegate(Type serviceType, IfUnresolved ifUnresolved) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 210
   at DryIoc.Container.DryIoc.IResolver.Resolve(Type serviceType, IfUnresolved ifUnresolved) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\DryIoc\Container.cs:line 195
   at Microsoft.Azure.WebJobs.Script.WebHost.DependencyInjection.ScopedServiceProvider.GetService(Type serviceType) in C:\azure-functions-host\src\WebJobs.Script.WebHost\DependencyInjection\ScopedServiceProvider.cs:line 25
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)
   at lambda_method(Closure , IServiceProvider , Object[] )
   at Microsoft.Azure.WebJobs.Host.Executors.DefaultJobActivator.CreateInstance[T](IServiceProvider serviceProvider) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\DefaultJobActivator.cs:line 37
   at Microsoft.Azure.WebJobs.Host.Executors.DefaultJobActivator.CreateInstance[T](IFunctionInstanceEx functionInstance) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\DefaultJobActivator.cs:line 32
   at Microsoft.Azure.WebJobs.Host.Executors.ActivatorInstanceFactory`1.<>c__DisplayClass1_1.<.ctor>b__0(IFunctionInstanceEx i) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\ActivatorInstanceFactory.cs:line 20
   at Microsoft.Azure.WebJobs.Host.Executors.ActivatorInstanceFactory`1.Create(IFunctionInstanceEx functionInstance) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\ActivatorInstanceFactory.cs:line 26
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2.CreateInstance(IFunctionInstanceEx functionInstance) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionInvoker.cs:line 44
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ParameterHelper.Initialize() in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 846
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsyncCore(IFunctionInstanceEx functionInstance, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 117
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsyncCore(IFunctionInstanceEx functionInstance, CancellationToken cancellationToken)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsync(IFunctionInstance instance, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 87
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsync(IFunctionInstance instance, CancellationToken cancellationToken)
   at Microsoft.Azure.WebJobs.Host.Executors.ShutdownFunctionExecutor.TryExecuteAsync(IFunctionInstance instance, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\ShutdownFunctionExecutor.cs:line 25
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Host.Executors.ShutdownFunctionExecutor.TryExecuteAsync(IFunctionInstance instance, CancellationToken cancellationToken)
   at Microsoft.Azure.WebJobs.JobHost.CallAsyncCore(IFunctionDefinition function, Object functionKey, IDictionary`2 arguments, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\JobHost.cs:line 241
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.JobHost.CallAsyncCore(IFunctionDefinition function, Object functionKey, IDictionary`2 arguments, CancellationToken cancellationToken)
   at Microsoft.Azure.WebJobs.JobHost.CallAsync(String name, IDictionary`2 arguments, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\JobHost.cs:line 229
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.JobHost.CallAsync(String name, IDictionary`2 arguments, CancellationToken cancellationToken)
   at Microsoft.Azure.WebJobs.Script.WebHost.Features.FunctionExecutionFeature.ExecuteAsync(HttpRequest request, CancellationToken cancellationToken) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Features\FunctionExecutionFeature.cs:line 67
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Features.FunctionExecutionFeature.ExecuteAsync(HttpRequest request, CancellationToken cancellationToken)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.FunctionInvocationMiddleware.GetResultAsync(HttpContext context, IFunctionExecutionFeature functionExecution) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\FunctionInvocationMiddleware.cs:line 140
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.FunctionInvocationMiddleware.GetResultAsync(HttpContext context, IFunctionExecutionFeature functionExecution)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.FunctionInvocationMiddleware.Invoke(HttpContext context) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\FunctionInvocationMiddleware.cs:line 61
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.FunctionInvocationMiddleware.Invoke(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.Middleware.DefaultMiddlewarePipeline.<>c.<<BuildPipeline>b__7_0>d.MoveNext() in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\JobHostMiddleware\DefaultMiddlewarePipeline.cs:line 31
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.Middleware.DefaultMiddlewarePipeline.<>c.<BuildPipeline>b__7_0(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.Middleware.HstsConfigurationMiddleware.<>c.<<-ctor>b__1_0>d.MoveNext() in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HstsConfigurationMiddleware.cs:line 24
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.Middleware.HstsConfigurationMiddleware.<>c.<.ctor>b__1_0(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.Middleware.HstsConfigurationMiddleware.Invoke(HttpContext context, RequestDelegate next) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HstsConfigurationMiddleware.cs:line 42
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.Middleware.HstsConfigurationMiddleware.Invoke(HttpContext context, RequestDelegate next)
   at Microsoft.Azure.WebJobs.Script.Middleware.DefaultMiddlewarePipeline.<>c__DisplayClass8_1.<<GetMiddlewareDelegate>b__1>d.MoveNext() in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\JobHostMiddleware\DefaultMiddlewarePipeline.cs:line 45
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.Middleware.DefaultMiddlewarePipeline.<>c__DisplayClass8_1.<GetMiddlewareDelegate>b__1(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.CustomHttpHeadersMiddleware.Invoke(HttpContext context, RequestDelegate next) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\CustomHttpHeadersMiddleware.cs:line 34
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.CustomHttpHeadersMiddleware.Invoke(HttpContext context, RequestDelegate next)
   at Microsoft.Azure.WebJobs.Script.Middleware.DefaultMiddlewarePipeline.<>c__DisplayClass8_1.<<GetMiddlewareDelegate>b__1>d.MoveNext() in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\JobHostMiddleware\DefaultMiddlewarePipeline.cs:line 45
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.Middleware.DefaultMiddlewarePipeline.<>c__DisplayClass8_1.<GetMiddlewareDelegate>b__1(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.JobHostPipelineMiddleware.Invoke(HttpContext httpContext, IJobHostMiddlewarePipeline middleware) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\JobHostPipelineMiddleware.cs:line 26
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.JobHostPipelineMiddleware.Invoke(HttpContext httpContext, IJobHostMiddlewarePipeline middleware)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.ResponseContextItemsCheckMiddleware.Invoke(HttpContext context) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\ResponseContextItemsCheckMiddleware.cs:line 29
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.ResponseContextItemsCheckMiddleware.Invoke(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HttpThrottleMiddleware.Invoke(HttpContext httpContext, IOptions`1 httpOptions, HttpRequestQueue requestQueue, HostPerformanceManager performanceManager, IMetricsLogger metricsLogger) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HttpThrottleMiddleware.cs:line 66
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HttpThrottleMiddleware.Invoke(HttpContext httpContext, IOptions`1 httpOptions, HttpRequestQueue requestQueue, HostPerformanceManager performanceManager, IMetricsLogger metricsLogger)
   at lambda_method(Closure , Object , HttpContext , IServiceProvider )
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.AspNetCore.Builder.UseWhenExtensions.<>c__DisplayClass0_1.<UseWhen>b__1(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HomepageMiddleware.Invoke(HttpContext context) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HomepageMiddleware.cs:line 34
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HomepageMiddleware.Invoke(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.ExceptionMiddleware.Invoke(HttpContext context) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\ExceptionMiddleware.cs:line 27
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.AppServiceHeaderFixupMiddleware.Invoke(HttpContext httpContext) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\AppServiceHeaderFixupMiddleware.cs:line 35
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.AppServiceHeaderFixupMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.ScriptHostRequestServiceProviderMiddleware.Invoke(HttpContext httpContext, WebJobsScriptHostService manager) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\ScriptHostRequestServiceProviderMiddleware.cs:line 30
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.ScriptHostRequestServiceProviderMiddleware.Invoke(HttpContext httpContext, WebJobsScriptHostService manager)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HostWarmupMiddleware.Invoke(HttpContext httpContext) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HostWarmupMiddleware.cs:line 34
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HostWarmupMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HostAvailabilityCheckMiddleware.Invoke(HttpContext httpContext, IScriptHostManager scriptHostManager) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HostAvailabilityCheckMiddleware.cs:line 47
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HostAvailabilityCheckMiddleware.Invoke(HttpContext httpContext, IScriptHostManager scriptHostManager)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.AspNetCore.Builder.UseWhenExtensions.<>c__DisplayClass0_1.<UseWhen>b__1(HttpContext context)
   at Microsoft.AspNetCore.Builder.UseWhenExtensions.<>c__DisplayClass0_1.<UseWhen>b__1(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.EnvironmentReadyCheckMiddleware.Invoke(HttpContext httpContext, IScriptWebHostEnvironment webHostEnvironment) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\EnvironmentReadyCheckMiddleware.cs:line 29
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.EnvironmentReadyCheckMiddleware.Invoke(HttpContext httpContext, IScriptWebHostEnvironment webHostEnvironment)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass4_1.<UseMiddleware>b__2(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HostnameFixupMiddleware.Invoke(HttpContext context) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\HostnameFixupMiddleware.cs:line 24
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.HostnameFixupMiddleware.Invoke(HttpContext context)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.SystemTraceMiddleware.Invoke(HttpContext context) in C:\azure-functions-host\src\WebJobs.Script.WebHost\Middleware\SystemTraceMiddleware.cs:line 47
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.Azure.WebJobs.Script.WebHost.Middleware.SystemTraceMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.HostFiltering.HostFilteringMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Hosting.HostingApplication.ProcessRequestAsync(Context context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequestsAsync[TContext](IHttpApplication`1 application)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequestsAsync[TContext](IHttpApplication`1 application)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.HttpConnection.ProcessRequestsAsync[TContext](IHttpApplication`1 httpApplication)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.HttpConnection.ProcessRequestsAsync[TContext](IHttpApplication`1 httpApplication)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.HttpConnectionMiddleware`1.OnConnectionAsync(ConnectionContext connectionContext)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.KestrelConnection.ExecuteAsync()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.KestrelConnection.ExecuteAsync()
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.KestrelConnection.System.Threading.IThreadPoolWorkItem.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

I have no knowledge about a different behavior for multiple constructor. I was able to keep a single constructor in my projects. I found no documentation or else to this topic.

Reply:
Wow.

> The Stack Trace does contain not much:

On the contrary, everything became clear. Azure host indeed uses different DI - DryIoc https://github.com/dadhi/DryIoc/issues/99

Here is some docs about ctor selection:
https://bitbucket.org/dadhi/dryioc/wiki/SelectConstructorOrFactoryMethod

Also azure host somehow configures this DI container. In short, the behavior is different from the "usual" `Microsoft.Extensions.DependencyInjection`. You can use any option for connecting dependencies, `builder.Services.AddSingleton<IDocumentExecuter>(sp => new DocumentExecuter());` will be OK for me. The main thing is to mention in the comments about this feature of using DryIoc and give a link to the documentation.

Reply:
Very Interesting with DryIoc. I added a comment.

(With not so much I only meant the Stack Trace of the Object not found.)